code/modules/events/zombie_outbreak.dm

```diff
@@ -0,0 +1,31 @@
+/datum/round_event_control/zombie_outbreak
+	name = "Zombie Outbreak"
+	typepath = /datum/round_event/zombie_outbreak
+	weight = 5
+	max_occurrences = 1
+	min_players = 30
+
+/datum/round_event/zombie_outbreak
+	fakeable = FALSE
+
+/datum/round_event/zombie_outbreak/start()
+	// The amount of infections among the crew depends on the amount of living, non-AFK, non-silicon players on the station
+	var/list/living_crew = list()
+	living_crew = get_living_station_crew()
+	var/infection_count = round(length(living_crew) / 10)
+
+	for(var/i in 0 to infection_count)
+		for(var/mob/living/carbon/human/H in shuffle(GLOB.player_list))
+			if(H.stat == DEAD)
+				continue
+			if(!SSjob.GetJob(H.mind.assigned_role) || (H.mind.assigned_role in GLOB.nonhuman_positions))
+				continue
+			if(!H.getorgan(/obj/item/organ/brain))
+				continue
+			if(!(MOB_ORGANIC in H.mob_biotypes))
+				continue
+			if(!H.getorganslot(ORGAN_SLOT_ZOMBIE))
+				var/obj/item/organ/zombie_infection/ZI = new()
+				ZI.Insert(H)
+			announce_to_ghosts(H)
+			break
```

Comment:
Rather than do 2 embedded loops and shuffling twice, instead only call break when infection count is less than 0, and subtract infection count by 1 each time
```suggestion
	for(var/mob/living/carbon/human/H in shuffle(GLOB.player_list))
		if(H.stat == DEAD)
			continue
		if(!SSjob.GetJob(H.mind.assigned_role) || (H.mind.assigned_role in GLOB.nonhuman_positions))
			continue
		if(!H.getorgan(/obj/item/organ/brain))
			continue
		if(!(MOB_ORGANIC in H.mob_biotypes))
			continue
		if(!H.getorganslot(ORGAN_SLOT_ZOMBIE))
			var/obj/item/organ/zombie_infection/ZI = new()
			ZI.Insert(H)
		announce_to_ghosts(H)
		infection_count  --
		if (infection_count <= 0)
			break
```

Reply:
done